// frontend/screens/homepage.styles.ts
import { StyleSheet, Dimensions } from 'react-native';

const { height: windowHeight } = Dimensions.get('window');

const styles = StyleSheet.create({
    safeArea: {
      flex: 1,
      backgroundColor: '#F9FAFB',
    },
    container: {
      flex: 1,
      position: 'relative',
    },
    headerContainer: {
      paddingVertical: 24,
      paddingHorizontal: 16,
      backgroundColor: '#FFFFFF',
      borderBottomWidth: 1,
      borderBottomColor: '#E5E7EB',
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.05,
      shadowRadius: 10,
      elevation: 2,
    },
    header: {
      fontSize: 28,
      fontWeight: '700',
      color: '#1F2937',
      textAlign: 'center',
      marginTop: 8,
      marginBottom: 4,
    },
    subheader: {
      fontSize: 16,
      color: '#6B7280',
      textAlign: 'center',
      marginBottom: 16,
    },
    progressBarContainer: {
      marginTop: 8,
    },
    progressBar: {
      height: 8,
      backgroundColor: '#E5E7EB',
      borderRadius: 4,
      marginBottom: 8,
      overflow: 'hidden',
    },
    progressFill: {
      height: '100%',
      backgroundColor: '#3B82F6',
      borderRadius: 4,
    },
    counterText: {
      fontSize: 14,
      color: '#3B82F6',
      fontWeight: '600',
      textAlign: 'right',
    },
    listContainer: {
      flex: 1,
    },
    listContentContainer: {
      paddingHorizontal: 16,
      paddingTop: 16,
    },
    coalitionGroup: {
      marginBottom: 24,
    },
    coalitionHeader: {
      fontSize: 18,
      fontWeight: '600',
      color: '#111827',
      marginBottom: 12,
      paddingBottom: 8,
      borderBottomWidth: 1,
      borderBottomColor: '#E5E7EB',
    },
    candidateItem: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: '#FFFFFF',
      padding: 16,
      marginBottom: 12,
      borderRadius: 16,
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 1 },
      shadowOpacity: 0.05,
      shadowRadius: 4,
      elevation: 2,
    },
    selectedCandidateItem: {
      borderWidth: 2,
      borderColor: '#3B82F6',
      backgroundColor: '#EFF6FF',
    },
    candidateInfo: {
      marginLeft: 12,
      flex: 1,
    },
    candidateName: {
      fontSize: 16,
      fontWeight: '500',
      color: '#111827',
      marginBottom: 4,
    },
    candidateMeta: {
      flexDirection: 'row',
      alignItems: 'center',
      marginTop: 4,
    },
    partyBadge: {
      backgroundColor: '#DBEAFE',
      paddingHorizontal: 8,
      paddingVertical: 2,
      borderRadius: 12,
      marginRight: 12,
    },
    candidateParty: {
      fontSize: 12,
      color: '#3B82F6',
      fontWeight: '500',
    },
    candidatePosition: {
      fontSize: 14,
      color: '#6B7280',
    },
    stickyFooter: {
      position: 'absolute',
      left: 0,
      bottom: 0,
      width: '100%',
      backgroundColor: 'rgba(249, 250, 251, 0.95)',
      paddingHorizontal: 16,
      paddingTop: 12,
      paddingBottom: 60,
      borderTopWidth: 1,
      borderTopColor: '#E5E7EB',
      shadowColor: '#000',
      shadowOffset: { width: 0, height: -2 },
      shadowOpacity: 0.1,
      shadowRadius: 8,
      elevation: 8,
    },
    submitButton: {
      backgroundColor: '#3B82F6',
      paddingVertical: 16,
      borderRadius: 12,
      alignItems: 'center',
      shadowColor: '#3B82F6',
      shadowOffset: { width: 0, height: 4 },
      shadowOpacity: 0.2,
      shadowRadius: 8,
      elevation: 4,
    },
    disabledButton: {
      backgroundColor: '#CBD5E0',
      shadowColor: 'transparent',
    },
    submitText: {
      color: 'white',
      fontWeight: '600',
      fontSize: 16,
    },
    
    // Modal Styles
    modalOverlay: {
      flex: 1,
      backgroundColor: 'rgba(0, 0, 0, 0.5)',
      justifyContent: 'center',
      alignItems: 'center',
    },
    modalContent: {
      width: '90%',
      maxHeight: '80%',
      backgroundColor: 'white',
      borderRadius: 20,
      padding: 24,
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 4 },
      shadowOpacity: 0.25,
      shadowRadius: 20,
      elevation: 5,
    },
    modalHeader: {
      fontSize: 22,
      fontWeight: '700',
      color: '#111827',
      marginBottom: 8,
      textAlign: 'center',
    },
    modalSubtext: {
      fontSize: 16,
      color: '#6B7280',
      marginBottom: 16,
      textAlign: 'center',
    },
    modalCandidateList: {
      maxHeight: windowHeight * 0.4,
      marginBottom: 16,
    },
    modalCandidateItem: {
      paddingVertical: 12,
      borderBottomWidth: 1,
      borderBottomColor: '#E5E7EB',
    },
    modalCandidateName: {
      fontSize: 16,
      fontWeight: '500',
      color: '#111827',
      marginBottom: 4,
    },
    modalCandidateDetails: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    modalPartyBadge: {
      backgroundColor: '#DBEAFE',
      paddingHorizontal: 8,
      paddingVertical: 2,
      borderRadius: 12,
      marginRight: 12,
    },
    modalPartyText: {
      fontSize: 12,
      color: '#3B82F6',
      fontWeight: '500',
    },
    modalPositionText: {
      fontSize: 14,
      color: '#6B7280',
    },
    modalButtonGroup: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginTop: 8,
    },
    modalCancelButton: {
      flex: 1,
      backgroundColor: '#F3F4F6',
      paddingVertical: 14,
      borderRadius: 12,
      alignItems: 'center',
      marginRight: 8,
    },
    modalCancelText: {
      color: '#4B5563',
      fontWeight: '600',
      fontSize: 16,
    },
    modalConfirmButton: {
      flex: 1,
      backgroundColor: '#3B82F6',
      paddingVertical: 14,
      borderRadius: 12,
      alignItems: 'center',
      marginLeft: 8,
      shadowColor: '#3B82F6',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.2,
      shadowRadius: 4,
      elevation: 2,
    },
    modalConfirmText: {
      color: 'white',
      fontWeight: '600',
      fontSize: 16,
    },
  });
export default styles;
